### DISCUSSION TITLE
Standardize/simplify the localize code

### DISCUSSION THREAD

**Discussion by rajkeshwar** [ing-bank/lion#2431] [Show and tell]
I recently submitted a PR that introduces a utility method for resolving the supplied locale and its fallback. However, as @tlouisse explained, the current verbosity in the code serves a purpose, and the team is considering standardizing or simplifying the localization logic.

To align with the team's direction, I’m starting this discussion to gather input on the approach being considered.

You can view my PR here: [#2430](https://github.com/ing-bank/lion/pull/2430).

Looking forward to your thoughts!

**Comment by gerjanvangeest** [comment]
As a first step, to keep the code the same but get rid of the verbosity, we plan to move all localize logic to a specific file [localizeNamespaceLoader.js](https://github.com/ing-bank/lion/blob/master/packages/ui/components/input-tel/src/localizeNamespaceLoader.js).

So that we only have a little bit of code left in each component:

```js
static localizeNamespaces = [
    { 'lion-input-tel': localizeNamespaceLoader },
    ...super.localizeNamespaces,
  ];
```

Is this something you could help us with @rajkeshwar.

**Reply by rajkeshwar** [reply]
Hi @gerjanvangeest sure I would love to do this. I believe the content of `localizeNamespaceLoader` is same for all the components except the path. So we can generate this file. I wrote a script to generate the `ocalizeNamespaceLoader.js` file.

Please have a look at this draft PR [2477](https://github.com/ing-bank/lion/pull/2477). I will cover all the scenarios and tests if the approach looks good.
